src/index.ts

```diff
@@ -2,250 +2,361 @@ import * as acorn from "acorn";
 import * as walk from "acorn-walk";
 import acornJSX from "acorn-jsx";
 import {
-  existsSync,
-  mkdirSync,
-  readdirSync,
-  readFileSync,
-  writeFileSync,
-} from "fs";
+	outputFileSync as writeFileSync,
+	existsSync,
+	mkdirSync,
+	readFileSync,
+} from "fs-extra";
+import { globSync } from "glob";
 import path from "path";
-import { CallExpression, MemberExpression, Node } from "estree";
+import {
+	Program,
+	CallExpression,
+	MemberExpression,
+	SequenceExpression,
+	UnaryExpression,
+	FunctionDeclaration,
+	ExpressionStatement,
+	Literal,
+	VariableDeclarator,
+	Node,
+	Identifier,
+	ImportDeclaration,
+} from "estree";
 import * as recast from "recast";
+import * as prettier from "prettier";
+
+import ClassParser from "./ClassParser";
+import CallParser from "./CallParser";
+import {
+	isCustomJSXRuntime,
+	parseComponentProps,
+	isBooleanOperator,
+	pushWithoutDuplicates,
+} from "./utils";
+import ImportParser from "./ImportParser";
+import ExportParser from "./ExportParser";
 
 const parser = acorn.Parser.extend(acornJSX());
 
-const files = readdirSync(path.join(process.cwd(), "input")).filter((f) =>
-  f.endsWith(".js")
-);
+if (!existsSync(path.join(process.cwd(), "input"))) {
+	mkdirSync(path.join(process.cwd(), "input"));
+}
 
 if (!existsSync(path.join(process.cwd(), "output"))) {
-  mkdirSync(path.join(process.cwd(), "output"));
+	mkdirSync(path.join(process.cwd(), "output"));
 }
 
+const files = globSync(`input/**/*.js`).map((file) =>
+	file.split("\\").slice(1).join("\\")
+);
+
 for (const file of files) {
-  // if (file !== "1445.js") continue; // DEBUG
-  console.log(file);
-  const filePath = path.join(process.cwd(), "input", file);
-  const data = readFileSync(filePath, "utf-8");
-  const ast = parser.parse(data, {
-    sourceType: "module",
-    ecmaVersion: "latest",
-  });
-  writeFileSync(
-    path.join(process.cwd(), "output", `${file}x`),
-    recast.print(parseModule(ast)).code
-  );
+	try {
+		console.log(file);
+		const filePath = path.join(process.cwd(), "input", file);
+		const data = readFileSync(filePath, "utf-8");
+		const ast = parser.parse(data, {
+			sourceType: "module",
+			ecmaVersion: "latest",
```

Comment:
Fixes the lack of parens on `var t = (Ae++).toString()`
```suggestion
			ecmaVersion: "latest",
			preserveParens: true,
```